app/backend/approaches/retrievethenread.py

```diff
@@ -165,9 +204,18 @@ async def run(
                 ),
             ],
         }
-
         completion = {}
-        completion["message"] = chat_completion["choices"][0]["message"]
+        if self.use_hugging_face:
+            completion["message"] = {
+                "content": chat_completion[0]['generated_text'],
+                "role": "assistant",
+                "function_call": None,
+                "tool_calls": None
+            }
+        else:
+            completion["message"] = chat_completion["choices"][0]["message"]
+
         completion["context"] = extra_info
         completion["session_state"] = session_state
+        
```

Comment:
General note: Please don't mix irrelevant changes with your PR changes. This is just a new line, but actual changes could be buried this way, when included in irrelevant PRs.